src/main/java/org/jabref/gui/importer/actions/OpenDatabaseAction.java

```diff
@@ -210,4 +216,33 @@ private LibraryTab addNewDatabase(ParserResult result, final Path file, boolean
         frame.addTab(libraryTab, raisePanel);
         return libraryTab;
     }
+
+    /* The layout to display in the tab when it's loading*/
+    public Node createLoadingLayout() {
+        ProgressIndicator progressIndicator = new ProgressIndicator(ProgressIndicator.INDETERMINATE_PROGRESS);
+        BorderPane pane = new BorderPane();
+        pane.setCenter(progressIndicator);
+
+        return pane;
+    }
+
+    public void onDatabaseLoadingStarted(LibraryTab libraryTab, BackgroundTask<?> backgroundTask) {
+        Node loadingLayout = createLoadingLayout();
+        libraryTab.setContent(loadingLayout);
+        libraryTab.setOnCloseRequest(e -> backgroundTask.cancel());
+
+        frame.addTab(libraryTab, true);
+    }
+
+    public void onDatabaseLoadingSucceed(LibraryTab libraryTab, ParserResult result) {
+        BibDatabaseContext context = result.getDatabaseContext();
+        libraryTab.feedData(context);
+
+        OpenDatabaseAction.performPostOpenActions(libraryTab, result);
+    }
+
+    public void onDatabaseLoadingFailed(LibraryTab libraryTab, Exception ex) {
+        dialogService.showErrorDialogAndWait(Localization.lang("Connection error"),
```

Comment:
It would be cook if you could pass the exception as paramter to the error dialog as well, it's just one addtional paramter